### PR TITLE
Change password

### PR DESCRIPTION
**Pull request by Oluane**

### CODE REVIEW

**Review by laurentmahieu — DISMISSED** [review]

**Review by laurentmahieu — DISMISSED** [review]

File: backend/route/user.js [hunk 1]
```diff
@@ -77,4 +78,47 @@ router.get("/:id/vehicle/nextmaintenance", (req, res) => {
...
+        return res.status(500).send("Error while fetching user!");
+      } else {
+        const isPrevPwMatch = bcrypt.compareSync(
+          formData.prevPw,
+          results[0].password
+        );
+        if (isPrevPwMatch) {
+          bcrypt.hash(formData.newPw, parseInt(saltRounds), (err, hash) => {
+            if (err) {
+              console.error(err);
```
Comment (LoicBrassart) [inline backend/route/user.js]: No console.log

File: backend/route/user.js [hunk 2]
```diff
@@ -77,4 +78,47 @@ router.get("/:id/vehicle/nextmaintenance", (req, res) => {
...
+            }
+
+            formData.newPw = hash;
+            connection.query(
+              `UPDATE USER SET password = ? WHERE id = ?`,
+              [formData.newPw, userId],
+              (err, results) => {
+                if (err) {
+                  // Si une erreur est survenue, alors on informe l'utilisateur de l'erreur
+                  console.error("Failure! " + err);
```
Comment (LoicBrassart) [inline backend/route/user.js]: No console.log

**Review by LoicBrassart — DISMISSED** [review]
Go, go, go ! :gun:

File: frontend/src/components/CarCard.jsx [hunk 3]
```diff
@@ -2,6 +2,7 @@ import React, { useEffect } from "react";
 import "./style/CarCard.scss";
 
 import { Link } from "react-router-dom";
+
```
Comment (laurentmahieu) [inline frontend/src/components/CarCard.jsx]: why?

File: frontend/src/App.js [hunk 4]
```diff
@@ -8,6 +8,7 @@ import Signup from "./pages/Signup";
 import HomePage from "./components/HomePage";
 import KmUpdate from "./components/KmUpdate";
 import AddCar from "./pages/AddCar";
+import ChangePassword from "./pages/ChangePassword";
```
Comment (laurentmahieu) [inline frontend/src/App.js:11]: import but never call?

**Review by laurentmahieu — APPROVED** [review]

**Review by LoicBrassart — APPROVED** [review]
SnE

**Pull request merged**